test/parallel/test-stream-backpressure.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const stream = require('stream');
+const { inspect } = require('util');
+
+let n = 0;
+const total = 65500 + 40 * 1024;
+const rs = new stream.Readable({
+  read: common.mustCall(function() {
+    if (n++ === 10) {
+      this.push(null);
+      return;
+    }
+
+    const length = this._readableState.length;
+
+    console.log(length, total);
+    assert(length < 4 * total);
```

Comment:
This is not going to be possible, as we are over highWaterMark here in half of the cases even on Node 8. I'll make the check stricter with `length <= total`, because that's what will happen.